docs/docs/porting-a-jekyll-website-to-gatsby.md

```diff
@@ -0,0 +1,276 @@
+
+## What changes need to be made?
+
+In order to transition your codebase over to using Gatsby, a few things need to be taken care of to account for the differences between how the projects are set up. First of all Gatsby want to know where to read `markdown` files by using the plugin [`gatsby-source-filesystem`](https://www.gatsbyjs.org/packages/gatsby-source-filesystem/) `( source plugin for sourcing data into your Gatsby application)` but how does it understand markdown format for this we need to use the plugin [`gatsby-transformer-remark`](https://www.gatsbyjs.org/packages/gatsby-transformer-remark/) to transform markdown files into something like gatsby can understand. This plugin is the responsible to read `markdown` files a extract information like the `frontmatter`, the excerpt and convert the `text` into `HTML` ready to be rendered.
+
+<!-- The plugin gatsby-transformer-remark is really usefull and also has many plugins to use within it.   -->
+
+## Development environment
+
+Gatsby generates websites and web applications for production through a compilation and build process, and it also has tools optimized for local development. To set up the Gatsby [CLI](/docs/glossary#cli) and development environment (if you haven't already) check out [Part Zero of the Gatsby tutorial](/tutorial/part-zero/).
+
+## Transition to Gatsby
+
+Gatsby can help you set up an application and removes much of the configuration headache. However, Gatsby offers some additional advantages like performance optimizations with static rendering and a thriving ecosystem of plugins, To get things set up and running we need to structure our jekyll-site files like this.
+
+```
+jekyll-site
+  ├── _includes
+  │    ├── footer.html
+  │    └── header.html
+  ├── _layouts
+  │    ├── default.html
+  │    └── post.html
+  ├── _posts
+  │    ├── 2007-10-29-why-every-programmer-should-play-nethack.md
+  │    └── 2009-04-26-barcamp-boston-4-roundup.md
+  ├── _sass
+  ├── _drafts
+  │    ├── begin-with-the-crazy-ideas.md
+  │    └── on-simplicity-in-technology.md
+  ├── _site
+  │    ├── about.md
+  │    └── blah-blah.md
+  ├── script
+  ├── .jekyll-metadata
+  ├── _config.yml
+  └── index.md
+```
+
+### Migrating the markdown / Liquid templates pages
+
+Create a new folder in Gatsby project under the `src/pages-markdown` for those pages written in markdown format. I left the `src/pages` as default so in future we need it for some javaScript files. Now its time to transfer all markdown files to `src/pages-markdown` like this.
+
+```
+  src
+  └── pages-markdown
+      ├── footer.md
+      └── header.md
+```
+
+In addition, these pages must be have next properties within the `frontmatter` section:
+
+```md
+---
+layout: page
+title: About
+date: 2010-11-23 14:48
+path: /about
+---
+```
+
+Now its time to make Gatsby engine take account into these files to inform where to read them:
```

Comment:
```suggestion
Once the markdown files are in the right place and include the necessary frontmatter, you'll need to configure your project so that it knows where to find them. Update your `gatsby-config.js` file with the appropriate `gatsby-source-filesystem` configuration.
```